### PR TITLE
Simple gemm dl

### PR DESCRIPTION
**Pull request by hsadasiv**
Created GEMV splitk for FP16 using atomicadd. 
Disclaimer: when testing out the example, you will need to modify kernel run to just once because of memset happening before the first atomicadd.

### CODE REVIEW

File: script/cmake-ck-dev.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ cmake
 -save-temps=$PWD"                                                                                 \
 -D CMAKE_BUILD_TYPE=Release                                                                       \
 -D BUILD_DEV=ON                                                                                   \
--D GPU_TARGETS="gfx908;gfx90a;gfx940"                                                             \
+-D GPU_TARGETS="gfx1100"                                                             \
```
Comment (zjing14) [inline script/cmake-ck-dev.sh]: Do not change this.

File: example/49_gemv_splitK/run_splitK_gemv_example.inc [hunk 2]
```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+struct ProblemSize_gemv final
+{
+    // ck::index_t M = 1; // 3840;
```
Comment (zjing14) [inline example/49_gemv_splitK/run_splitK_gemv_example.inc]: Remove these comments.

File: example/49_gemv_splitK/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#arg2: initialization (0=no init, 1=integer value, 2=decimal value)
+#arg3: run kernel # of times (>1)
+#arg4: number of splitK batches
+./bin/example_gemm_xdl 0 1 5 151
+
+```
+
+Result (MI100 @ 1087Mhz, 133.5TFlops peak FP16)
+```
+a_m_k: dim 2, lengths {3840, 4096}, strides {4096, 1}
```
Comment (zjing14) [inline example/49_gemv_splitK/README.md]: Why test Gemm instead of GemmV?

File: example/49_gemv_splitK/common.hpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+#include "ck/library/utility/device_memory.hpp"
+#include "ck/library/utility/fill.hpp"
+#include "ck/library/utility/host_tensor.hpp"
+#include "ck/library/utility/host_tensor_generator.hpp"
+#include "ck/library/utility/literals.hpp"
+#include "ck/library/reference_tensor_operation/cpu/reference_gemm.hpp"
+
+struct ProblemSize final
+{
+    ck::index_t M = 3840;
```
Comment (zjing14) [inline example/49_gemv_splitK/common.hpp]: We could use a GemmV example here. Fixed M = 1.

**Comment by zjing14** [comment]
```
$ ./bin/example_splitK_gemv_fp16 1 1 1 2
a_m_k: dim 2, lengths {1, 4608}, strides {4608, 1}
b_k_n: dim 2, lengths {4608, 1104}, strides {1, 4608}
c_m_n: dim 2, lengths {1, 1104}, strides {1104, 1}
Perf: 0.140048 ms, 0.0726499 TFlops, 72.7315 GB/s, deviceGemvDl<64, 1, 128, 4, 8, 1, 2, 1>
Error: Incorrect results!        out[0] != ref[0]: inf != 39104
Error: Incorrect results!        out[1] != ref[1]: 4564 != 414
Error: Incorrect results!        out[2] != ref[2]: 10352 != 942
Error: Incorrect results!        out[3] != ref[3]: -913 != -83
   max err: inf
```
You can do verification before timing like this
```
    c_m_n_device_buf.Zero();

    invoker.Run(argument, StreamConfig{nullptr, false});

    if(config.do_verification)
    {
       ...
    }

    float ave_time = invoker.Run(argument, StreamConfig{nullptr, config.time_kernel});
```

File: example/49_gemv_splitK/README.md (hunk 3 above)
Comment (hsadasiv) [reply example/49_gemv_splitK/README.md]: I will update README.

File: example/49_gemv_splitK/common.hpp (hunk 4 above)
Comment (hsadasiv) [reply example/49_gemv_splitK/common.hpp]: Sure, I'll move my ProblemSize_gemv from run_splitK_gemv_example.inc file and replace the one here.

**Comment by zjing14** [comment]
Please rename '`splitK_gemmv`' to '`gemmv_splitk`' for consistency.

**Comment by hsadasiv** [comment]
@zjing14 
I have made the suggested revisions, merged and resolved conflicts. However, I see the static checks /clang-format failing in the checks. How do I fix this? 
Right now I am using the clang-format extension from vscode. Maybe it missed some files. Do you have a recommendation on how to fix this?

**Comment by hsadasiv** [comment]
Changes moved to new branch: PR#1028

**Pull request closed without merging**